dome/install_ubuntu_src.md

```diff
@@ -0,0 +1,292 @@
+
+Some tools require Python 3.5 (or higher) which is not the default option on some
+platforms (like Ubuntu Focal). The Python
+[virtualenv](https://virtualenv.pypa.io/en/latest/) could be a useful solution in
+cases where the default option cannot be easily changed.
+
+## vcstool and colcon from pip
+
+PIP is available on all platforms:
+
+```bash
+pip install vcstool
+```
+
+```bash
+pip install -U colcon-common-extensions
+```
+
+Check that no errors were printed while installing with PIP. If your system is not recognising the commands, and you're using a system that is compatible with Debian or Ubuntu packages, see the instructions below to install using `apt`.
+
+## vcstool and colcon from apt
+
+An alternative method is to use the `.deb` packages available on Debian or Ubuntu:
+
+```bash
+sudo sh -c 'echo "deb http://packages.ros.org/ros/ubuntu $(lsb_release -sc) main" > /etc/apt/sources.list.d/ros-latest.list'
+sudo -E apt-key adv --keyserver hkp://ha.pool.sks-keyservers.net:80 --recv-key C1CF6E31E6BADE8868B172B4F42ED6FBAB17C654
+sudo apt-get update
+sudo apt-get install python3-vcstool python3-colcon-common-extensions
+```
+
+## Git
+
+Ignition libraries use `git` for version control, so it must be available
+in the system for `vcstool` to work properly.
+
+```bash
+sudo apt-get install git
+```
+
+## Getting the sources
+
+The instructions below use some UNIX commands to manage directories but the
+equivalent alternatives on Windows should provide the same result.
+
+The first step is to create a developer workspace in which `vcstool` and
+`colcon` can work:
+
+```bash
+mkdir -p ~/workspace/src
+cd ~/workspace/src
+```
+
+All the sources of ignition-dome are declared in a yaml file. Download
+it to the workspace:
+
+```bash
+wget https://raw.githubusercontent.com/ignition-tooling/gazebodistro/master/collection-dome.yaml
+```
+
+Use `vcstool` to automatically retrieve all the Ignition libraries sources from
+their repositories:
+
+```bash
+vcs import < collection-dome.yaml
+```
+
+The src subdirectory should contain all the sources ready to be built.
+
+## Install dependencies
+
+Before compiling it is necessary to install all the dependencies of the different
+packages that compose the Dome collection. Every platform has a different
+method to install software dependencies.
+
+Add `packages.osrfoundation.org` to the apt sources list:
+
+```bash
+sudo sh -c 'echo "deb http://packages.osrfoundation.org/gazebo/ubuntu-stable `lsb_release -cs` main" > /etc/apt/sources.list.d/gazebo-stable.list'
+wget http://packages.osrfoundation.org/gazebo.key -O - | sudo apt-key add -
+sudo apt-get update
+```
+
+The command below will install all dependencies in Ubuntu Focal:
+
+```bash
+sudo apt-get install -y \
+  cmake \
+  freeglut3-dev \
+  libavcodec-dev \
+  libavdevice-dev \
+  libavformat-dev \
+  libavutil-dev \
+  libbenchmark-dev \
+  libcurl4-openssl-dev \
+  libdart-collision-ode-dev \
+  libdart-dev \
+  libdart-utils-urdf-dev \
```

Comment:
For a build to succeed for me on Ubuntu Focal, I also had to install:

* `libdart-external-ikfast-dev`
* `libdart-external-odelcpsolver-dev`